app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt

```diff
@@ -76,20 +81,43 @@ class AutoCompleteApi @Inject constructor(
         if (query.isBlank()) {
             return Observable.just(AutoCompleteResult(query = query, suggestions = emptyList()))
         }
-
-        val savedSitesObservable = getAutoCompleteBookmarkResults(query)
-            .zipWith(
-                getAutoCompleteFavoritesResults(query),
-            ) { bookmarks, favorites ->
-                (favorites + bookmarks).take(2)
-            }
+        val savedSitesObservable: Observable<List<AutoCompleteSuggestion>> =
+            getAutoCompleteBookmarkResults(query)
+                .zipWith(
+                    getAutoCompleteFavoritesResults(query),
+                ) { bookmarks, favorites ->
+                    (favorites + bookmarks)
+                }.map {
+                    it.sortedByDescending { it.score }.mapNotNull {
+                        val savedSite = it.savedSite
+                        AutoCompleteBookmarkSuggestion(
+                            phrase = savedSite.url.toUri().toStringDropScheme(),
+                            title = savedSite.title,
+                            url = savedSite.url,
+                            isAllowedInTopHits = savedSite is Favorite,
```

Comment:
This is redundant now, but will be needed later on, as bookmarks will be allowed in top hits in case they are a duplicate of a history item